tiledb/sm/array_schema/dimension.cc

```diff
@@ -1533,7 +1533,16 @@ Status Dimension::set_null_tile_extent_to_range() {
 
   // Calculate new tile extent equal to domain range
   auto domain = (const T*)domain_.data();
-  T tile_extent = domain[1] - domain[0];
+  T tile_extent;
+
+  // Calculate the ratio of tile extent to max value of T
+  double tile_extent_ratio = (1.0 * domain[1]) / std::numeric_limits<T>::max() -
+                             (1.0 * domain[0]) / std::numeric_limits<T>::max();
+  if (tile_extent_ratio > 0.1) {
+    tile_extent = std::numeric_limits<T>::max() / 10;
```

Comment:
This doesn't work at all. Note the overflow check below. Note also the comment at original-line 1546 below: "After this, tile_extent = domain[1] - domain[0] + 1, which is the correct domain range". This is as much documentation as we've got at present. The postcondition of the function should satisfy this predicate. That means that if `domain[1] - domain[0]` won't fit in the type we're given, we must return an error.

The problem is that this means detecting an overflow, which can be tricky. For unsigned types (all are integral), we can't overflow on the subtraction, even if we can on the increment below. For signed integral types, overflow is undefined behavior, so you have to test for overflow without triggering overflow. For floating point types, overflow becomes NaN, which has a marker value.

Reply:
Thanks for the comments, we can keep the default tile_extent range to be domain[1]-domain[0]+1, if it is overflow, just return error message. Actually I am thinking about the benefits of using a such large tile_extent. If we can set some limits(I prefer to a constant), it will be much easier to handle it. I am still confused why this 
 double tile_extent_ratio = (1.0 * domain[1]) / std::numeric_limits&lt;T&gt;::max() -                             (1.0 * domain[0]) / std::numeric_limits&lt;T&gt;::max();" will trigger overflow. Could you give me more details or some references? thanks

Reply:
The inescapable problem is the divide-by-10. The postcondition is that the extent is equal to the difference in the dimensions in all cases, not just the convenient ones.

As for overflow, casting to float can lose accuracy. If the type is int64_t, in floating point terms that's 62 bits of mantissa. A double only has 53 bits of mantissa. So converting to double can increase the value on a round-up. Or more to the point here, round down so that you end up with a false negative on an overflow test.

In general, when you're testing for overflow, you can't practically use type conversions, since they all seem to have boundary cases that don't behave like you'd want.

Reply:
when end users try to create a dimension like this Dimension::create<int64_t>(ctx, "rows", {{-9223372036854775806, 9223372036854775807}}), I think what we should to do is to provide an appropriate default tile_extent for them, since this is a valid dimension for int64_t. I really hesitate to return an error status for that. Of course we don't need to calculate an optimized tile extent according to cpu cache size or other things which will make things complicated. I will wait for more comments about that

Reply:
Please look at my pull request... I already have a fix for this out.

Reply:
thanks, then I will delete this branch. I just randomly found the ticket 7282 when I was waiting for reviews of other branches.